### PR TITLE
fix(rule): prevent duplicate-index state from leaking across lint sessions

### PR DESCRIPTION
**Pull request by bolajiwahab**
## Background and Motivation

This PR addresses a state-leak bug in the GN025 “DuplicateIndex” rule by moving index-tracking state off the class and adding a regression test to prevent future regressions.

## Summary of Changes

- Update GN025 to store `seen_indexes` on the checker instance instead of as a class-level variable.
- Add a regression test intended to ensure duplicate-index tracking doesn’t leak beyond its intended scope.
## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Housekeeping (change that does not affect functionality)

## SQL Examples (if applicable)

```sql
-- Before
...

-- After
...
```

## Checklist

- [x] I have read and followed the [contributing guidelines](https://github.com/bolajiwahab/pgrubic/blob/main/docs/docs/contributing.md)
- [x] I have checked that there are no other open [Pull Requests](https://github.com/bolajiwahab/pgrubic/pulls) for the same update/change
- [x] I have performed a self-review of my code
- [x] I have added/updated tests (positive + negative cases)
- [ ] I have updated documentation (if applicable)

### CODE REVIEW

File: src/pgrubic/rules/general/GN025.py [hunk 1]
```diff
@@ -28,7 +26,10 @@ class DuplicateIndex(linter.BaseChecker):
     Remove the duplicate.
     """
 
-    seen_indexes: typing.ClassVar[list[typing.Any]] = []
+    def __init__(self, *, config: config.Config) -> None:
+        """Initialize the DuplicateIndex checker."""
+        super().__init__(config=config)
+        self.seen_indexes: list[tuple[object, object, object, object]] = []
```
[the remark this answers is not in the document]
Comment (bolajiwahab) [reply src/pgrubic/rules/general/GN025.py:32]: Not per linter run but per session, we want to check previous and next files for duplicate indexes in a session.

File: tests/test_rules.py [hunk 2]
```diff
@@ -88,3 +88,27 @@ def test_rules(
...
+        config = core.parse_config()
+        linter = core.Linter(config=config, formatters=core.load_formatters)
+        for rule in core.load_rules(config=config, include_deprecated=True):
+            linter.checkers.add(rule(config=config))
+
+        linting_result = linter.run(source_file="test.sql", source_code=sql)
+
+        assert not any(
+            violation.rule_code == "GN025" for violation in linting_result.violations
+        ), "Test failed: index flagged as duplicate across independent Linter instances"
```
[the remark this answers is not in the document]
Comment (bolajiwahab) [reply tests/test_rules.py:114]: Not per linter run but per session, we want to check previous and next files for duplicate indexes.
